indico/modules/rb/client/js/modules/calendar/CalendarListView.jsx

```diff
@@ -101,14 +130,90 @@ class CalendarListView extends React.Component {
           {moment(day, 'YYYY-MM-DD').format('dddd, LL')}
         </Header>
         <Card.Group itemsPerRow={4} stackable styleName="day-cards-container">
-          {bookings.map(this.renderBooking)}
+          {bookings.map(booking => this.renderBooking(booking, day))}
         </Card.Group>
         <Divider hidden />
       </div>
     );
   };
 
-  renderBooking = booking => {
+  renderLink = (booking, day) => {
+    const {
+      linkData,
+      isAdminOverrideEnabled,
+      actions: {linkBookingOccurrence},
+    } = this.props;
+    if (!linkData) {
+      return;
+    }
+    const {linkingConfirm} = this.state;
+    const {reservation} = booking;
+    const boundaries = [moment(linkData.startDt), moment(linkData.endDt)];
+    const datesMatch =
+      moment(booking.startDt).isBetween(...boundaries, undefined, '[]') &&
+      moment(booking.endDt).isBetween(...boundaries, undefined, '[]');
```

Comment:
I think this check may be too strict. IMHO it's reasonable to allow linking a booking that "contains" the event - e.g. when the booking is from 8:30-11:00 and the event is from 9:00-10:30.

Reply:
cc @OmeGak

Reply:
And isn't that the case with the current code?

Reply:
Nope, the current logic requires that both conditions are met:

- the occurrence start is between the start/end date of the event
- the occurrence end is between the start/end date of the event

However, I think what we need is a check that:

- the occurrence start is <= the event start
- the occurrence end is >= the event end

Reply:
Or maybe even have a general "time ranges overlap" check since you may want to link multiple occurrences that are used for the event even though neither of them fully contains the event?

That would be `(StartOccurrence < EndEvent) and (EndOccurrence > StartEvent)`

Reply:
That does not work. Imagine the following:

Event: 10::00 - 12:00.
Occurrence: 11:00 - 16:00.

Using your formula:
11:00 < 12:00 and 16:00 > 10:00

Yet between 10:00 and 11:00 there's an event but there's no booking.